RecoTracker/DisplacedRegionalTracking/plugins/DisplacedRegionProducer.h

```diff
@@ -0,0 +1,62 @@
+#ifndef RecoTracker_DisplacedRegionalTracking_DisplacedRegionProducer_h
+#define RecoTracker_DisplacedRegionalTracking_DisplacedRegionProducer_h
+
+#include <list>
+#include <vector>
+#include <limits>
+#include <string>
+#include <atomic>
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/Candidate/interface/VertexCompositeCandidate.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/View.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+#include "RecoTracker/DisplacedRegionalTracking/interface/DisplacedRegion.h"
+
+struct DisplacedRegionTFCache {
+  DisplacedRegionTFCache() : graphDef(nullptr) {}
+
+  std::atomic<tensorflow::GraphDef *> graphDef;
```

Comment:
Correct, `atomic` is not needed here (it really seems to be used in place of `mutable`). I'd actually suggest
```suggestion
  std::unique_ptr<tensorflow::GraphDef> graphDef;
```
to avoid the need for explicit `delete`. I'd further suggest to move the Session to the cache (since it is nowadays thread safe), make both graph and Session private, and provide public accessor to return a pointer (or reference) to const Session. Or, could the entire module be made `global` at that point? (by quick look it would look feasible)